File: wedding-backend/app/api/notifications.py
from __future__ import annotations

from typing import Optional

from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.database import get_db
from app.middleware.auth import get_current_user
from app.models.user import User
from app.schemas.notification import NotificationReadRequest
from app.services.notification_service import NotificationService

router = APIRouter()


@router.get("")
async def list_notifications(
    is_read: Optional[bool] = Query(None),
    type: Optional[str] = Query(None),
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    service = NotificationService(db)
    return await service.list_notifications(current_user.id, is_read, type, page, page_size)


@router.get("/unread-count")
async def unread_count(
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    service = NotificationService(db)
    count = await service.get_unread_count(current_user.id)
    return {"count": count}


@router.put("/read")
async def mark_read(
    data: NotificationReadRequest,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    service = NotificationService(db)
    await service.mark_as_read(current_user.id, data.ids)
    await db.commit()
    return {"message": "ok"}


@router.put("/read-all")
async def mark_all_read(
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    service = NotificationService(db)
    await service.mark_all_as_read(current_user.id)
    await db.commit()
    return {"message": "ok"}

File: wedding-frontend/src/api/notifications.ts
import request from './index'

export interface Notification {
  id: number
  title: string
  content: string
  type: string
  is_read: boolean
  related_id: number | null
  related_type: string | null
  created_at: string | null
}

export function getNotifications(params?: {
  is_read?: boolean
  type?: string
  page?: number
  page_size?: number
}): Promise<any> {
  return request.get('/notifications', { params })
}

export function getUnreadCount(): Promise<{ count: number }> {
  return request.get('/notifications/unread-count')
}

export function markAsRead(ids: number[]): Promise<any> {
  return request.put('/notifications/read', { ids })
}

export function markAllAsRead(): Promise<any> {
  return request.put('/notifications/read-all')
}
